### PR TITLE
Adds copy buttons to website, and better descriptions to API Docs

### PR DESCRIPTION
**Pull request by aliabd**
- Adds copy buttons to all codeblocks on the website (docs, guides, landing page, demos, etc) Closes: #5210 
- Adds missing docstring to HighlightedText Closes: #5277 
- Adds description to return types in Api Docs, and uses better language for filepaths Closes:  #4917

Before & after for api docs 

Screen Shot 2023-09-27 at 4 31 39 PM

Screen Shot 2023-09-27 at 4 22 56 PM

### CODE REVIEW

File: client/python/gradio_client/serializing.py [hunk 1]
```diff
@@ -152,7 +152,10 @@ class ImgSerializable(Serializable):
     """Expects a base64 string as input/output which is serialized to a filepath."""
 
     def serialized_info(self):
-        return {"type": "string", "description": "filepath or URL to image"}
+        return {
+            "type": "string",
+            "description": "filepath on your computer, or URL to image",
```
Comment (abidlabs) [inline client/python/gradio_client/serializing.py:157]: Is this clearer? 
```suggestion
            "description": "filepath on your computer (or URL) of image",
```

Comment (abidlabs) [reply client/python/gradio_client/serializing.py:157]: Similarly for the others

File: gradio/components/highlighted_text.py [hunk 2]
```diff
@@ -52,6 +52,7 @@ def __init__(
         """
         Parameters:
             value: Default value to show. If callable, the function will be called whenever the app loads to set the initial value of the component.
+            color_map: A dictionary mapping labels to colors. The colors must be specified as hex codes.
```
Comment (abidlabs) [inline gradio/components/highlighted_text.py:55]: I don't think they must be specified as hex codes actually. Would be good to maybe give an example, e.g. `{"person": "red", "location": "#FFEE22"}

**Comment by abidlabs** [comment]
@aliabd there's a small issue with the API docs, which is that part of the comment overflows to the next line. Specifically the:

`in 'x' Audio component`

When you copy the code, it turns out to be invalid because it copies the comment as code. This is what gets copied:


```
from gradio_client import Client

client = Client("http://127.0.0.1:7868/")
result = client.predict(
				"https://github.com/gradio-app/gradio/raw/main/test/test_files/audio_sample.wav",	# str (filepath on your computer, or URL to file)
								in 'x' Audio component
				api_name="/predict"
)
print(result)
```

Could we fix this?

**Comment by abidlabs** [comment]
Otherwise, looks good to me. Feels very nice to have copy buttons back again!

**Comment by aliabd** [comment]
Thanks for reviewing @abidlabs. Fixed up everything, please take another look.

File: js/app/src/api_docs/CodeSnippets.svelte [hunk 3]
```diff
@@ -64,19 +64,16 @@ result = client.predict(<!--
 				-->{/if}<!--
 			--><span class="desc"
 								><!--
-			-->	# {python_type.type} {#if python_type.description}({python_type.description})
-								{/if}<!--
-			-->in '{label}' <!--
+			-->	# {python_type.type} {#if python_type.description}({python_type.description}){/if}<!---->in '{label}' <!--
```
Comment (abidlabs) [inline js/app/src/api_docs/CodeSnippets.svelte:67]: There should be a space before the "in":

image

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Tiny nit, otherwise lgtm! Thanks @aliabd for the nice fixes

File: js/app/src/api_docs/CodeSnippets.svelte (hunk 3 above)
Comment (aliabd) [reply js/app/src/api_docs/CodeSnippets.svelte:67]: oops! missed that

**Pull request merged**